### DISCUSSION TITLE
Document/Create Error 500 Docker Container Unwritable

### DISCUSSION THREAD

**Discussion by alkaris2** [open-webui/open-webui#1730] [General]
# Bug Report

## Description

I'm getting Internal Server Error 500 when attempting to create or add Modelfiles to Open WebUI. Error 500 is displayed, as Docker container is not writable by default, so it will not allow create or import of Modelfiles.

WebUI also seems to not understand Modelfiles that don't have JSON file type extension, but also unable to read the file when JSON is affixed to the file name.

> [!NOTE]
>Make this easily consistent on access. `Modelfile.json or Modelfile` and have it know what the file is even if it doesn't have `.json` extension on the filename, especially when you download it from https://openwebui.com and it doesn't have a file extension when you Download manually.

**Bug Summary:**
Attempt to Import or Manually add Modelfiles, and get Error 500 when it can't write file inside the docker container.

**Steps to Reproduce:**
Copied commands from documentation;
```
docker run -d -p 3000:8080 --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main
```

**Expected Behavior:**
Create and import Modelfiles without issue.

**Actual Behavior:**
Internal Server Error 500 - Cannot write to Docker container.

## Environment

- **Open WebUI Version:** v0.1.120
[(latest)](https://github.com/open-webui/open-webui/releases/tag/v0.1.120)

- **Operating System:** Manjaro Linux 23.1.3
- **Browser (if applicable):** Firefox 124.01

## Reproduction Details

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs.

## Logs and Screenshots

![image](https://github.com/open-webui/open-webui/assets/73155407/bdce01a7-23dd-451c-8b9d-f4f7d2425d10)

## Installation Method

Docker install setup

**Comment by liubin595338764** [comment]
<img width="949" alt="image" src="https://github.com/open-webui/open-webui/assets/5274428/4761ea6a-79ea-4074-b565-df02700460a0">
我可以上传txt文件，不能上传md文件。而且引用txt文件来提问就会报错，普通用文字提问时正常的

**Comment by tjbck** [comment]
Unable to reproduce the issue on my end using `Get` button. Also, it seems like your webui isn't connected to Ollama? As for the import feature, importing downloaded file won't work for now as they're designed to work Ollama directly, but I'll see what can be done once I have time.

**Reply by liubin595338764** [reply]
ollama是正常运行的，你看这个请求是正常的
<img width="268" alt="image" src="https://github.com/open-webui/open-webui/assets/5274428/e489dd50-b631-4632-ad54-b140be7ac595">
